test/EndToEndTests/Tests/Core/Microsoft.OData.Core.E2E.Tests/WriteJsonPayloadTests/WritePayloadTests.cs

```diff
@@ -0,0 +1,1132 @@
+                var outputWithoutModel2 = rex.Replace(outputWithoutModel, "");
+                TestsHelper.VerifyPayloadString(outputWithModel2, outputWithoutModel2, mimeType);
+            }
+            else if (mimeType == MimeTypes.ApplicationJson + MimeTypes.ODataParameterMinimalMetadata)
+            {
+                var rex = new Regex("\"\\w*@odata.type\":\"#[\\w\\(\\)\\.]*\",");
+                var outputWithoutModel2 = rex.Replace(outputWithoutModel, "");
+                TestsHelper.VerifyPayloadString(outputWithModel, outputWithoutModel2, mimeType);
+            }
+            else
+            {
+                //NoMetadata with/out model should result in same output
+                Assert.Equal(outputWithModel, outputWithoutModel);
+            }
+        }
+
+        /// <summary>
+        /// Write an entry containing stream, named stream
+        /// </summary>
+        [Theory]
+        [InlineData(MimeTypeODataParameterFullMetadata)]
+        [InlineData(MimeTypeODataParameterMinimalMetadata)]
+        [InlineData(MimeTypeODataParameterNoMetadata)]
+        public async Task CarEntryTest(string mimeType)
+        {
+            var settings = new ODataMessageWriterSettings
+            {
+                ODataUri = new ODataUri() { ServiceRoot = _baseUri },
+                EnableMessageStreamDisposal = false
+            };
+            string outputWithModel = null;
+            string outputWithoutModel = null;
+
+            var responseMessageWithModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithModel.SetHeader("Content-Type", mimeType);
+            responseMessageWithModel.PreferenceAppliedHeader().AnnotationFilter = "*";
+
+            var carType = _model.FindDeclaredType(NameSpacePrefix + "Car") as IEdmEntityType;
+            var carSet = _model.EntityContainer.FindEntitySet("Cars");
+
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithModel, settings, _model))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceWriterAsync(carSet, carType);
+                outputWithModel = await WriteAndVerifyCarEntryAsync(responseMessageWithModel, odataWriter, true, mimeType);
+            }
+
+            var responseMessageWithoutModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithoutModel.SetHeader("Content-Type", mimeType);
+            responseMessageWithoutModel.PreferenceAppliedHeader().AnnotationFilter = "*";
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceWriterAsync();
+                outputWithoutModel = await WriteAndVerifyCarEntryAsync(responseMessageWithoutModel, odataWriter, false,mimeType);
+            }
+
+            TestsHelper.VerifyPayloadString(outputWithModel, outputWithoutModel, mimeType);

```

Comment:
What does this method do?

Reply:
It asserts if the `outputWithModel` and `outputWithoutModel` are equal. I've gotten rid of it.